### PR TITLE
post test failure status and link test run url on github for windows tests

### PR DESCRIPTION
**Pull request by rohit21agrawal**
this is a required change to make some improvements to test reporting in the GitHub status.

This makes sure that each test publish task only has to publish to one test URL which can be linked to in the GitHub status .

Since dotnet vstest runner only outputs test results in the TRX format, I have used a test logger adapter to output results in XUnit xml format in this PR.

Summary:

Use the VSTS Test Run API to obtain the test run URL for each of the test phase, and use that url to update the git commit status. It also finds the number of test failures and adds that to the description in case any tests failed. If tests did not run or error'd out before the results could even be published, then the linked URL will be the build url.

Right now, this is only being done for windows based machines as i am waiting for our IT team to install powershell on the linux and mac machines, and then we will be able to use the same scripts across all OS.

CC: @rrelyea

### CODE REVIEW

**Review by emgarten — APPROVED** [review]
👏 looks good, it will be great to have the list on github

**Review by PatoBeltran — APPROVED** [review]

**Comment by rohit21agrawal** [comment]
@emgarten i have updated the title and description of the PR to reflect the extra work i have done after your approval.

**Pull request merged**
